### DISCUSSION TITLE
Safari 15 has ResizeObserver API but it still does not work

### DISCUSSION THREAD

**Discussion by 205g0** [ZeeCoder/use-resize-observer#112] [Q&A] [answered]
Pretty much the title and according to https://caniuse.com/resizeobserver and other sources the ResizeObserver API landed in Safari on iOS 2020, so no poyfill is needed. My code works on Chrome on Firefox flawlessly.

**Comment by ZeeCoder** [comment]
Hi, can you provide a minimal reproduction of the issue, ideally on codesandbox?

I'm testing the library in Safari 13 via Browserstack, so it should work.

**Comment by 205g0** [comment]
I tested on a real iPhone and with latest Safari (15). Maybe you get somehow an iPhone first (e.g., from a friend) or at least the simulator within XCode on a Mac? Once you have one of either _and_ running latest Safari , I'm happy to provide a min repo. Providing anything before while having different testing envs doesn't make too much sense...

{btw, I am not an Apple guy either, I just have years old Apple hardware lying around for testing purposes, e.g., the first iPhone SE is avail for less than 100 bucks refurbed and runs latest iOS and Safari)

**Comment by 205g0** [comment]
Hi again, I quickly made a min repo, here you go, https://codesandbox.io/s/ancient-river-zb0z5?file=/src/App.tsx

The direct link just to the view on an iPhone is avail under https://zb0z5.csb.app/ but pls check first the first link to understand the source.

The view should show 'Hello Code Sandbox [this div's width size]' but shows on Safari 15 'Hello Code Sandbox 1'

**Comment by 205g0** [comment]
@ZeeCoder could you reproduce the error?

**Comment by 205g0** [comment]
@ZeeCoder are you okay? Do you need anything more I can provide?

**Comment by ZeeCoder** [comment]
Hey, sorry I can't look at it yet.
Earliest would be the weekend but we'll see.

**Comment by 205g0** [comment]
Ok, thanks for letting me know and no worries! 🙂

**Comment by ZeeCoder** [comment]
I'll keep you posted 👍

**Comment by ZeeCoder** [comment, marked as answer]
Seems like Safari does not support the box property as advertised [here](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver/observe#browser_compatibility).
The properties are [not actually available](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry#browser_compatibility), which is why you're not getting a size report when you ask for `border-box`.
(`content-box` should work fine.)

This really highlights how experimental all this is, and support is weird too.

For example, in Chrome / FF, RO will actually respond with **all** box sizes, regardless of what you're actually asking for. :man_shrugging: 
![image](https://user-images.githubusercontent.com/3679506/137646029-10bb01f7-0d71-4957-80dc-a016d2b1a795.png)

I might be mistaken, but I think this behaviour is actually against the spec.

You can test browser support by running this little code snippet:
```js
new ResizeObserver((entries) => {
	console.log(entries[0])
}).observe(document.body, {
	box: 'border-box'
});
```

**Comment by ZeeCoder** [comment]
I could probably highlight this in the readme, although I'm already linking to the same support tables.

**Comment by 205g0** [comment]
Hey thanks for looking into it! Meanwhile I migrated to Svelte, so I don't need this anymore but still thanks for helping out!

**Comment by ZeeCoder** [comment]
no worries, have fun hacking 👌

**Comment by vincerubinetti** [comment]
Safari is driving me absolutely crazy. Apparently, just to let people know, `ResizeObserver` doesn't seem to work on `<svg>` elements, at least of Version 16.4 (18615.1.26.110.1)

On a webpage that has an SVG that scales with the window width, try this in dev tools:

```javascript
new ResizeObserver(console.log).observe($("svg"))
```

[jsfiddle](https://jsfiddle.net/7enksdbw/show)

This should console log when you resize the window. It works fine in Chrome and Firefox, but not Safari. If you change the element being observed to something else, like a `<div>`, then it starts to work in Safari.

Yet another example of Safari's horrible implementation of SVG.

[Related stackoverflow post](https://stackoverflow.com/questions/65565149/how-to-apply-resizeobserver-to-svg-element)
[Webkit bug report](https://bugs.webkit.org/show_bug.cgi?id=254078)

**Comment by vincerubinetti** [comment]
As of Safari 17.2.1 it seems like ResizeObserver is working on SVGs.

**Comment by ZeeCoder** [comment]
Safary 15 is basically not in use anymore, [and SVGs are handled by RO nowadays](https://github.com/ZeeCoder/use-resize-observer/discussions/94#discussioncomment-17852307).
